be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -292,21 +292,34 @@ ChunkPtr JsonScanner::_cast_chunk(const starrocks::vectorized::ChunkPtr& src_chu
 }
 
 JsonReader::JsonReader(starrocks::RuntimeState* state, starrocks::vectorized::ScannerCounter* counter,
-                       JsonScanner* scanner, std::shared_ptr<SequentialFile> file, bool strict_mode)
+                       JsonScanner* scanner, std::shared_ptr<SequentialFile> file, bool strict_mode,
+                       const std::vector<SlotDescriptor*>& slot_descs)
         : _state(state),
           _counter(counter),
           _scanner(scanner),
           _strict_mode(strict_mode),
           _file(std::move(file)),
-          _next_line(0),
-          _total_lines(0),
-          _closed(false) {
+          _slot_descs(slot_descs) {
 #if BE_TEST
     raw::RawVector<char> buf(_buf_size);
     std::swap(buf, _buf);
 #endif
 }
 
+Status JsonReader::open() {
+    RETURN_IF_ERROR(_read_and_parse_json());
+    _empty_parser = false;
+
+    if (_scanner->_json_paths.empty() && _scanner->_root_paths.empty()) {
+        // Since the iterating order may be different with json field when the json path/root is set,
+        // reorder the column descriptor as json field, when the json path and json root is empty.
```

Comment:
What if each json document has different order? Why not reorder the JSON according to the column-order?

Reply:
1. It may not get a good performance.
2. It seems no good way to modify the value of `simdjson::ondemand::object`.